client/app/pages/queries/components/QueryExecutionStatus.jsx

```diff
@@ -0,0 +1,67 @@
+import { includes } from "lodash";
+import React from "react";
+import PropTypes from "prop-types";
+import Alert from "antd/lib/alert";
+import Button from "antd/lib/button";
+import { Timer } from "@/components/Timer";
+
+export default function QueryExecutionStatus({ status, updatedAt, error, onCancel }) {
+  const alertType = status === "failed" ? "error" : "info";
+  const showTimer = status !== "failed" && updatedAt;
+  const canCancel = includes(["waiting", "processing"], status);
+  let message = null;
+
+  switch (status) {
+    case "waiting":
+      message = <React.Fragment>Query in queue&hellip;</React.Fragment>;
+      break;
+    case "processing":
+      message = <React.Fragment>Executing query&hellip;</React.Fragment>;
+      break;
+    case "loading-result":
+      message = <React.Fragment>Loading results&hellip;</React.Fragment>;
+      break;
+    case "failed":
+      message = (
+        <React.Fragment>
+          Error running query: <strong>{error}</strong>
+        </React.Fragment>
+      );
+      break;
+    // no default
+  }
+
+  return (
+    <Alert
```

Comment:
I think using [banner](https://ant.design/components/alert/#components-alert-demo-banner) styling may result in a better-looking alert, but perhaps it's worth it to change Antd `Alert` style globally.